walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,105 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Set;
 
+/**
+* Finds available times for a meeting.
+*/
 public final class FindMeetingQuery {
+  /** Find and returns collection of free time ranges for scheduling a meeting.*/
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() >= TimeRange.WHOLE_DAY.duration()) {
+      return new ArrayList<TimeRange>();
+    }
+    ArrayList<Event> allEvents = copyEvents(events);
```

Comment:
You might be able to remove this function all together
```suggestion
    ImmutableList<Event> allEvents = ImmutableList.copyOf(events);
```

(making immutable bc it's read only)

Another option..

```suggestion
    ArrayList<Event> allEvents = new ArrayList<>();
    events.forEach(event -> allEvents.add(event));
```

Reply:
Couldn't figure out how to get first option working so went with the second one for now. Will learn more and hopefully refactor.